### =========================================================================
### Single-cell RNA-seq Datasets
### -------------------------------------------------------------------------
###

meta <- data.frame(
  Title = c('pollen',
            'darmanis',
            'kolodziejczyk',
            'zeisel',
            'hipsc',
            'cellmix',
            'pbmc',
            'hca_bm',
            'hca_cb'),
  Description = c(paste0("The Pollen dataset was generated from 11 populations using",
                          "microfluidics. The authors analyzed single-cell transcriptomes",
                          "across downsampled sequencing depths, in order to demonstrate",
                          "that shallow single-cell mRNA sequencing (âˆ¼50,000 reads per cell)",
                          "is sufficient for unbiased cell-type classification and",
                          "biomarker identification."),
                  paste0("The Darmanis dataset was generated using cortical tissue",
                          "from eight adults and four embryonic samples ranging",
                          "from 16 to 18 gestational weeks in age."),
                  paste0("The Kolodziejczyk dataset was generated using mouse Embryonic",
                          "Stem Cells (mESCs) cultured in three different conditions:",
                          "serum, 2i, and the alternative ground state a2i"),
                  paste0("The Zeisel dataset was generated to classify cells in",
                          "the mouse somatosensory cortex and hippocampal CA1 region.",
                          "The authors found 47 molecularly distinct subclasses,",
                          "comprising all known major cell types in the cortex."),
                  paste0("The hiPSC dataset was generated using WTC-CRISPRi Human",
                          "Induced Pluripotent Stem Cells (hiPSCs), which are genetically",
                          "engineered with an inducible nuclease-dead Cas9 fused to a",
                          "KRAB repression domain."),
                  paste0("The Cellmix dataset was generated internally by AbbVie Inc.",
                          "The dataset was generated at two different AbbVie sites using",
                          "three Non-Small Cell Lung Cancer (NSCLC) cell lines, both",
                          "fresh and frozen."),
                  paste0("The PBMC dataset was generated by 10x Genomics using the",
                          "Cell Ranger 1.1.0 pipeline. The official name of the dataset",
                          "is `Fresh 68k PBMCs (Donor A)`."),
                  paste0("The hca-bm dataset was generated by Human Cell Atlas using",
                          "bone marrow cells from eight donors."),
                  paste0("The hca-cb dataset was generated by Human Cell Atlas using",
                          "cord blood cells")),
  BiocVersion = "0.1.0",
  Genome = c(rep("NA",5),"GRCh38", "hg19", "GRCh38", "GRCh38"),
  SourceType = rep("BAM", 9),
  SourceUrl = c("https://www.ncbi.nlm.nih.gov/sra?term=SRP041736",
                "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67835",
                "https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2600/",
                "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60361",
                "https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6687/",
                "NA",
                "https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/fresh_68k_pbmc_donor_a",
                "https://data.humancellatlas.org/explore/projects/cc95ff89-2e68-4a08-a234-480eca21ce79",
                "https://data.humancellatlas.org/explore/projects/cc95ff89-2e68-4a08-a234-480eca21ce79"),
  SourceVersion = c("Aug 02 2014",
                    "May 20, 2015",
                    "Jul 28 2017",
                    "Feb 20, 2015",
                    "Aug 17 2018",
                    "NA",
                    "Jul 24 2016",
                    "NA",
                    "NA"),
  Species = c(rep("Homo sapiens",2),
              rep("Mus musculus",2),
              rep("Homo sapiens",5)),
  TaxonomyId = c(rep(9606,2),
                 rep(10090,2),
                 rep(9606,5)),
  Coordinate_1_based = TRUE,
  DataProvider = c("SRA","GEO","EMBL-EBI","GEO","EMBL-EBI","AbbVie Inc.","10X Genomics","HCA","HCA"),
  Maintainer = "Wachter, A. <astrid.wachter@abbvie.com>",
  RDataClass = "Large SingleCellExperiment",
  DispatchClass = "Rda",
  RDataPath = paste0('benchmark.data.scRNAseq/',
                     c('pollen.rda',
                        'darmanis.rda',
                        'kolodziejczyk.rda',
                        'zeisel.rda',
                        'hipsc.rda',
                        'cellmix.rda',
                        'pbmc.rda',
                        'hca_bm.rda',
                        'hca_cb.rda')),
  Tags = "",
  Notes = "",
  Publication = c("https://www.nature.com/articles/nbt.2967",
                  "https://www.pnas.org/content/112/23/7285",
                  "https://www.ncbi.nlm.nih.gov/pubmed/26431182",
                  "https://science.sciencemag.org/content/347/6226/1138",
                  "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6028138/",
                  "",
                  "",
                  "https://www.sciencedirect.com/science/article/pii/S0301472X18308051",
                  "")
)

write.csv(meta, file="inst/extdata/metadata.csv", row.names=FALSE)
